File: in_Java/0414 Third Maximum Number.java
class Solution {
    public int thirdMax(int[] nums) {
        long a = Long.MIN_VALUE;
        long b = Long.MIN_VALUE;
        long c = Long.MIN_VALUE;
        for(int i : nums){
            if(i > a){
                c = b;
                b = a;
                a = i;
            }else if(i > b && i < a){
                c = b;
                b = i;
            }
            else if(i > c && i < b){
                c = i;
            }
        }
        return (int)(c == Long.MIN_VALUE? a : c);
    }
}

File: in_Python/0414 Third Maximum Number.py
class Solution(object):
    def thirdMax(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        one = two = three = -sys.maxint
        for i in nums:
            if i > one:
                one, two, three = i, one, two
            elif i < one and i > two:
                two, three = i, two
            elif i < two and i > three:
                three = i
        return three if three != -sys.maxint else one